scripts/reset-simulators.rb

```diff
@@ -17,55 +20,70 @@ def platform_for_device_type(device_type)
   end
 end
 
-def wait_for_core_simulator_service
-  # Run until we get a result since switching simulator versions often causes CoreSimulatorService to throw an exception.
-  while `xcrun simctl list devices`.empty?
-  end
-end
-
-def running_devices(devices)
-  devices.select { |device| device['state'] != 'Shutdown' }
-end
-
-def shutdown_simulator_devices(devices)
-  # Shut down any simulators that need it.
-  running_devices(devices).each do |device|
-    puts "Shutting down simulator #{device['udid']}"
-    system("xcrun simctl shutdown #{device['udid']}") or puts "    Failed to shut down simulator #{device['udid']}"
-  end
-end
-
 begin
+  # - Check if the wrong version of the CoreSimulatorService is running, stop it if so
+  if ENV['JENKINS_HOME'] || ! /^\d+\s+#{ENV['DEVELOPER_DIR']}\/.+/ =~ `/usr/bin/pgrep -lf com.apple.CoreSimulator.CoreSimulatorService 2>/dev/null` ; then    
+    # Kill all the current simulator processes as they are from a different Xcode version
```

Comment:
This appears to be an answer as to my other question as to why we're doing things conditionally, not to why we're changing the approach we're using to kill simulator-related processes.

I don't think 2 is a compelling reason. I've not heard any complaints from engineers on the Cocoa team about simulators being reset by this script. Our goal is to have local testing of changes via `build.sh` behave as closely as possible to testing in CI, and we should try and stick to that unless we have a good reason not to.